api/apis/app_handler.go

```diff
@@ -105,7 +100,8 @@ func (h *AppHandler) appGetHandler(authInfo authorization.Info, r *http.Request)
 
 	app, err := h.appRepo.GetApp(ctx, authInfo, appGUID)
 	if err != nil {
-		return nil, h.handleGetAppErr(err, appGUID)
+		h.logger.Error(err, "Failed to fetch app from Kubernetes", "AppGUID", appGUID)
+		return nil, apierrors.ForbiddenAsNotFound(err)
```

Comment:
The name of the helper function is somewhat confusing, as it makes it look like the code is expecting a forbidden error and is translating it to a not found error. The fact that other types of errors are passed-through without modification is lost without looking at the implementation of the function.